core/src/CMakeLists.txt

```diff
@@ -159,6 +159,7 @@ if(Kokkos_ENABLE_IMPL_MDSPAN)
     target_link_libraries(kokkoscore PUBLIC std::mdspan)
   elseif(KOKKOS_COMPILER_SUPPORTS_MDSPAN AND NOT Kokkos_ENABLE_IMPL_SKIP_COMPILER_MDSPAN)
     message(STATUS "Using compiler-supplied mdspan")
+    set(KOKKOS_IMPL_USE_COMPILER_PROVIDED_MDSPAN ON)
```

Comment:
What if the compiler supports `experimental/mdspan`?

Reply:
If the compiler provides `mdspan` header but the feature macro `__cpp_lib_mdspan` isn't defined to a value of at least 202207L, we are still silently using the bundled version AFAICT. We should include the logic in `Kokkos_MDSpan_Header` here, i.e.,
```
__cpp_lib_mdspan >= 202207L && !defined(KOKKOS_ENABLE_CUDA) && \
    !defined(KOKKOS_ENABLE_HIP)
```
and then only use `KOKKOS_IMPL_USE_COMPILER_PROVIDED_MDSPAN` in that file.

Reply:
ok that's a good point. I'll see what can be worked into that logic in cmake. I also need to fix the experimental/mdspan thing since nvhpc still ships with that